components/clarity-repl/src/analysis/coverage_tests.rs

```diff
@@ -721,10 +718,25 @@ fn multiple_test_files() {
         paths.insert(contract_id.name.to_string(), format!("/contract-{i}.clar"));
     }
 
-    let cov = coverage_hook.collect_lcov_content(&asts, &paths);
+    let cov = session.collect_lcov_content(&asts, &paths);
 
     assert_eq!(
         [
+            "TN:",
+            "SF:/contract-0.clar",
+            "FN:1,add",
+            "FNF:1",
+            "FNH:0",
+            "BRF:0",
+            "BRH:0",
+            "end_of_record",
+            "SF:/contract-1.clar",
+            "FN:1,add",
+            "FNF:1",
+            "FNH:0",
+            "BRF:0",
+            "BRH:0",
+            "end_of_record",
```

Comment:
As a sidev effect, this is the wanted behaviour for code coverage.
Because session.deploy_contract uses the session coverage hook.
This is therefore the initial analysis of the contracts.
If we wanted to pass the eval_hooks to `update_session_with_deployment_plan` ([here](https://github.com/hirosystems/clarinet/blob/6f178d8ca8a9f7a29db749c077214082462741ff/components/clarinet-sdk-wasm/src/core.rs#L455)) instead, this would make pass eval_hooks args in a whole lot more of functions (and it comes with some difficulties down the line)

Reply:
So this a wanted behavior for better coverage handling; It doesn't really impact code coverage % though